zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/ProcessInstanceControllerTest.java

```diff
@@ -779,4 +783,372 @@ void shouldRejectMigrateProcessInstanceWithOperationReferenceNotValid() {
+    assertThat(capturedRequest.processInstanceKey()).isEqualTo(1);
+    assertThat(capturedRequest.activateInstructions()).isEmpty();
+    assertThat(capturedRequest.terminateInstructions()).isNotEmpty();
+    assertThat(capturedRequest.terminateInstructions().size()).isEqualTo(2);
+    assertThat(capturedRequest.operationReference()).isEqualTo(123L);
+  }
+
+  @Test
+  void shouldModifyProcessInstanceWithoutTerminateInstructions() {
+    // given
+    when(processInstanceServices.modifyProcessInstance(any(ProcessInstanceModifyRequest.class)))
+        .thenReturn(CompletableFuture.completedFuture(new ProcessInstanceModificationRecord()));
+
+    final var request =
+        """
+        {
+          "activateInstructions": [
+            {
+              "elementId": "elementId",
+              "variableInstructions": [
+                {
+                  "variables": {
+                    "foo": "bar"
+                  }
+                }
+              ],
+              "ancestorElementInstanceKey": 123456
+            },
+            {
+              "elementId": "elementId2",
+              "ancestorElementInstanceKey": 654321
+            }
+          ],
+          "operationReference": 123
+        }""";
+
+    // when/then
+    webClient
+        .post()
+        .uri(MODIFY_PROCESS_URL.formatted("1"))
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isNoContent();
+
+    Mockito.verify(processInstanceServices).modifyProcessInstance(modifyRequestCaptor.capture());
+    final var capturedRequest = modifyRequestCaptor.getValue();
+    assertThat(capturedRequest.processInstanceKey()).isEqualTo(1);
+    assertThat(capturedRequest.activateInstructions()).isNotEmpty();
+    assertThat(capturedRequest.activateInstructions().size()).isEqualTo(2);
+    assertThat(capturedRequest.terminateInstructions()).isEmpty();
+    assertThat(capturedRequest.operationReference()).isEqualTo(123L);
+  }
+
+  @Test
+  void shouldRejectModifyProcessInstanceWithActivateInstructionsElementNull() {
+    // given
+    final var request =
+        """
+        {
+          "activateInstructions": [
+            {
+              "ancestorElementInstanceKey": 123456
+            },
+            {
+              "elementId": "elementId2",
+              "ancestorElementInstanceKey": 654321
+            }
+          ],
+          "terminateInstructions": [
+            {
+              "elementInstanceKey": 123456
+            },
+            {
+              "elementInstanceKey": 654321
+            }
+          ],
+          "operationReference": 123
+        }""";
+
+    final var expectedBody =
+        """
+        {
+            "type":"about:blank",
+            "title":"INVALID_ARGUMENT",
+            "status":400,
+            "detail":"At least one of [elementId, ancestorElementInstanceKey] is required.",
```

Comment:
❓ There is one of `[elementId, ancestorElementInstanceKey]` present in the request. Should this say "both elementId and ancestorElementInstanceKey are required"?

Reply:
Yes